core/src/main/java/tc/oc/pgm/filters/Filters.java

```diff
@@ -1,14 +1,42 @@
 package tc.oc.pgm.filters;
 
+import java.util.List;
 import javax.annotation.Nullable;
 import org.bukkit.block.Block;
 import org.bukkit.block.BlockState;
 import org.bukkit.util.BlockVector;
+import tc.oc.pgm.api.filter.Filter;
+import tc.oc.pgm.filters.modifier.QueryModifier;
 
-/** Utility methods useful for implementing filters */
+/** Utility methods useful for implementing and working with filters */
 public class Filters {
   private Filters() {}
 
+  public static List<Filter> flattenFilter(Filter filter, List<Filter> result) {
+    result.add(filter);
+
+    // No easier way to check for child filters
```

Comment:
Then maybe add one?

Consider just adding a `ParentFilter` interface or similar that just extends Filter, and adds a `getChildren` method
then simply call getChildren here recrusively if the filter is of type ParentFilter, and make sure that all filters (SingleFunctionFilter, MultiFilterFunction, and QueryModifier) all implement ParentFilter too and reply to getChildren with a stream or list of their filters, essentially, the logic you've done here, but with each having their own

You can even go a step further and just have that getChildren function be the recursive one itself, that way if you just call getChildren on the topmost filter you get all the child ones without needing to recurse yourself

Reply:
This method is not necessary because all filters can be found the feature definition context anyways